### PR TITLE
Updates to chat popup and GET HEAD

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* None

*Description of changes:*
- cleaned up chat-popup.vue
- don't render chat-popup if no url
- fixed tz in chat-popup
- removed common mixin from date-mixin
- wrapped "fetch head" in try-catch (re: https://sentry.io/organizations/registries/issues/2296940817/?project=5433950)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: src/components/common/chat-popup.vue [hunk 1]
```diff
@@ -1,17 +1,17 @@
 <template>
-  <div class="button-container">
-    <form
+  <div class="button-container" v-if="webChatUrl">
```
Comment (severinbeauvais) [inline src/components/common/chat-popup.vue:2]: Don't display button if web chat is not configured (eg, locally).

File: src/components/common/chat-popup.vue [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
+    <v-form
       id="webchat"
       target="webchat_window"
       method="post"
-      :action="window['webChatUrl']"
-      onSubmit="window.open('about:blank','webchat_window','width=400,height=500');"
+      :action="webChatUrl"
+      @submit="onSubmit()"
```
Comment (severinbeauvais) [inline src/components/common/chat-popup.vue:8]: I moved the `window` statements out of the template and into the script.

File: src/components/common/chat-popup.vue [hunk 3]
```diff
@@ -53,20 +53,31 @@ import axios from 'axios'
...
   chatError = false
+  chatStatus = ''
+
+  /** The user's browser's tz offset (not necessarily Pacific time). */
+  get timeZoneOffset (): number {
+    return new Date().getTimezoneOffset() / 60
+  }
+
+  async mounted (): Promise<void> {
+    if (this.webChatStatusUrl) {
```
Comment (severinbeauvais) [inline src/components/common/chat-popup.vue:68]: Don't fetch status if web chat is not configured (eg, locally).

File: src/mixins/date-mixin.ts [hunk 4]
```diff
@@ -24,21 +23,17 @@ export class DateMixin extends Mixins(CommonMixin) {
     const input = `${window.location.origin}${process.env.VUE_APP_PATH}/`
     const init: RequestInit = { cache: 'no-store', method: 'HEAD' }
 
-    // don't call fetch() during Jest tests
-    // because it's not defined
-    if (this.isJestRunning) return new Date()
```
Comment (severinbeauvais) [inline src/mixins/date-mixin.ts:29]: If fetch is undefined, the catch block executes.

**Review by cameron-freshworks — APPROVED** [review]
Looks good!

**Pull request merged**